src/common/args.cpp

```diff
@@ -685,6 +685,25 @@ std::string HelpMessageOpt(const std::string &option, const std::string &message
 
 const std::vector<std::string> TEST_OPTIONS_DOC{
     "addrman (use deterministic addrman)",
+    "fastprune (Use smaller block files and lower minimum prune height for testing purposes)",
+    "checkblocks (How many blocks to check at startup)",
+    "checklevel (How thorough the block verification is)",
+    "checkblockindex (Do a consistency check for the block tree, chainstate, and other validation data structures every <n> operations. Use 0 to disable.)",
+    "checkaddrman (Run addrman consistency checks every <n> operations. Use 0 to disable.)",
+    "checkmempool (Run mempool consistency checks every <n> transactions. Use 0 to disable.)",
+    "checkpoints Enable rejection of any forks from the known historical chain",
+    "deprecatedrpc (Allows deprecated RPC method(s) to be used)",
+    "stopafterblockimport (Stop running after importing blocks from disk )",
+    "stopatheight (Stop running after reaching the given height in the main chain)",
+    "limitancestorcount (Do not accept transactions if number of in-mempool ancestors is <n> or more)",
+    "limitancestorsize (Do not accept transactions whose size with all in-mempool ancestors exceeds <n> kilobytes)",
+    "limitdescendantcount (Do not accept transactions if any ancestor would have <n> or more in-mempool descendants)",
+    "limitdescendantsize (Do not accept transactions if any ancestor would have descendants exceeding <n> kilobytes)",
```

Comment:
From what I understand, the startup options `deprecatedrpc`, `stopatheight`, `limitancestorcount`, `limitancestorsize`, `limitdescendantcount`, and `limitdescendantsize` are definitely used in production by some users.

Reply:
In the arguments description in `init.cpp` they are categorised as `DEBUG_ONLY` AND `DEBUG_TEST` which is where I picked them up to be "test-only". There was not much documentation elsewhere to decide otherwise.

Reply:
I see. What’s the overarching intent here: is the purpose just to label them more explicitly as being for testing, or is the intent to restrict their use to test networks?

Reply:
We are restricting its use in test networks only specifically to `regtest` only . the `-test` flag has a check to assert use of `regtest`.

Reply:
Sorry, I have to withdraw `limitancestorcount`, `limitancestorsize`, `limitdescendantcount`, and `limitdescendantsize`. Both parties that I suspected to use these startup options, in fact, do not use them. We definitely used `stopatheight` at a prior employer while indexing the blockchain, and my understanding was that `deprecatedrpc` was meant to give users a heads-up that an RPC would be going away with the next release, but not completely requiring an immediate transitioning away from it.

Reply:
`stopatheight` is also frequently used for benchmarks on mainnet, most recently in #28280. The `-checkXYZ ` startup options that can help debug issues on all networks and I've run some of them on several occasions on mainnet and signet.

Many of the options are debug-only because they are meant to only be used by devs testing new things, but that doesn't mean they can only be useful regtest - often it's helpful to test things on other networks too.

I think that restricting to regtest should only be done with options specifically introduced for the test suite that don't make sense anywhere else.

Reply:
makes sense. moving regtest only options to `-test=<opt>` would make the interface(dev-testing/more-debug-info vs regtest-only) less confusing. (other than test=addrman), regtest-only would include: `fastprune`, `mocktime`, `acceptstalefeeestimates`.
I think `peertimeout` and `rpcdoccheck` are only used in tests, so you could include them too